src/commands/BootstrapCommand.js

```diff
@@ -100,150 +122,141 @@ export default class BootstrapCommand extends Command {
+   * Symlink package binaries to dependent packages' node_modules/.bin directory
+   * @param {Function} callback
+   */
+  symlinkPackages(callback) {
+    this.logger.info("Symlinking packages and binaries");
+    const actions = [];
+    this.filteredPackages.forEach((filteredPackage) => {
+      const packageLocation = path.join(this.repository.packagesLocation, filteredPackage.name);
+      Object.keys(filteredPackage.allDependencies)
+        // filter out external dependencies
+        .filter((dependency) => find(this.packages, { name: dependency}))
+        .forEach((dependency) => {
+          // skip this dependency?
+          let skip = false;
+          // get path to dependency and its package.json
+          const dependencyLocation = path.join(this.repository.packagesLocation, dependency);
+          const dependencyPackageJsonLocation = path.join(dependencyLocation, "package.json");
+          // ignore dependencies without a package.json file
+          if (!FileSystemUtilities.existsSync(dependencyPackageJsonLocation)) {
+            this.logger.error(`Unable to find package.json for ${dependency} dependency. Skipping...`);
+          } else {
+            const installedDepLocation = path.join(filteredPackage.nodeModulesLocation, dependency);
+            // check if dependency is already installed
+            if (FileSystemUtilities.existsSync(installedDepLocation)) {
+              const isDepSymlink = FileSystemUtilities.isSymlink(installedDepLocation);
+              // installed dependency is a symlink pointing to a different location
+              if (isDepSymlink !== dependencyLocation) {
+                this.logger.warning(
+                  `Symlink already exists for ${dependency} dependency of ${filteredPackage.name},` +
+                  "but links to different location. Replacing with updated symlink..."
+                );
+              // installed dependency is not a symlink
+              } else if (!isDepSymlink) {
+                // dependency is compatible with existing package, remove folder and replace with symlink
+                if (semver.satisfies(
+                    require(dependencyPackageJsonLocation).version,
+                    filteredPackage.allDependencies[dependency]
+                  )) {
+                  this.logger.warning(
+                    `${dependency} is already installed for ${filteredPackage.name}. ` +
+                    "Replacing with symlink..."
+                  );
+                  // remove installed dependency
+                  actions.push((cb) => FileSystemUtilities.rimraf(installedDepLocation, cb));
+                } else {
+                  // dependency is not compatible with existing package, do nothing
+                  this.logger.warning(
+                    `${dependency} is already installed for ${filteredPackage.name}. ` +
+                    "No compatible package found. Skipping..."
+                  );
```

Comment:
Wait... haven't we already installed repo deps that don't satisfy semver as external dependencies by the time we get here?  We should be able to just filter those out above.